app_build_suite/build_steps/helm.py

```diff
@@ -356,6 +372,11 @@ def initialize_config(self, config_parser: configargparse.ArgParser) -> None:
             action="store_true",
             help="Generate the metadata file for Giant Swarm App Platform.",
         )
+        config_parser.add_argument(
+            "--catalog-name",
+            required="--generate-metadata" in sys.argv,
```

Comment:
This is tricky, I suggest to change it. The config lib we're using seamlessly integrates config options coming from a config file, env variables and command line. They have their own priorities, but basically you can put them anywhere (so anything not changing across builds goes to the config file and the user or CI/CD can still override/add with env vars or command line).

So, just checking `sys.argv` will fail here. The best way is probably to make it optional here and validate if it is set (if needed) in `pre_run`.